numpy/core/include/numpy/npy_cpu.h

```diff
@@ -84,6 +84,8 @@
         # error Unknown ARM CPU, please report this to numpy maintainers with \
 	information about your platform (OS, CPU and compiler)
     #endif
+#elif defined(_M_ARM64)
+    #define NPY_CPU_ARMEL_AARCH64
```

Comment:
Could you not just get the other declaration with a few `||`?

Reply:
Did you mean to add the `_M_ARM64` to the code block starting at line 69 ? if so, the problem is none of the other endianness checking macros are available for MSVC and we will need to have more condition checks inside.

Reply:
`NPY_CPU_ARMEL_AARCH64` is first defined on 79.  You could just make the `_M_ARM64` path also hit that `#define` rather than having a second define for `NPY_CPU_ARMEL_AARCH64`

Reply:
Did you mean something like the following?

```
#elif defined(__arm__) || defined(__aarch64__) || defined(_M_ARM64)
    #if defined(__ARMEB__) || defined(__AARCH64EB__)
        #if defined(__ARM_32BIT_STATE)
            #define NPY_CPU_ARMEB_AARCH32
        #elif defined(__ARM_64BIT_STATE)
            #define NPY_CPU_ARMEB_AARCH64
        #else
            #define NPY_CPU_ARMEB
        #endif
    #elif defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
        #if defined(__ARM_32BIT_STATE)
            #define NPY_CPU_ARMEL_AARCH32
        #elif defined(__ARM_64BIT_STATE) || defined(_M_ARM64)
            #define NPY_CPU_ARMEL_AARCH64
        #else
            #define NPY_CPU_ARMEL
        #endif
    #else
        # error Unknown ARM CPU, please report this to numpy maintainers with \
	information about your platform (OS, CPU and compiler)
    #endif

```

Reply:
Yes.  I noticed every other target seems to only be defined once, so maintaining the pattern seems reasonable to me.